lib/srv/alpnproxy/proxy.go

```diff
@@ -212,10 +230,14 @@ func New(cfg ProxyConfig) (*Proxy, error) {
 	if err := cfg.CheckAndSetDefaults(); err != nil {
 		return nil, trace.Wrap(err)
 	}
+	cfg.IdentityTLSConfig.ClientAuth = tls.RequireAndVerifyClientCert
+	fn := auth.AddClusterCAPoolsToTLSConfig(cfg.IdentityTLSConfig, cfg.AccessPoint, cfg.ClusterName, cfg.Log)
+	cfg.IdentityTLSConfig.GetConfigForClient = fn
```

Comment:
Can this be done as a part of `CheckAndSetDefaults`? It already clones the config.